scripts/Test-Signatures.ps1

```diff
@@ -0,0 +1,209 @@
+    if($suppress) {
+        return;
+    }
+    
+    Clear-Line;
+    Write-Host "$('  ' * $Depth)[$operation] " -ForegroundColor Yellow -NoNewline
+    Write-Host $message -NoNewline
+
+    if ($newline -or $CIBuild) {
+        Write-Host;
+    }
+}
+
+function Submit-Operation($operation, $message, [switch]$NoNewLine, $suppress=$false) {
+    if($suppress) {
+        return;
+    }
+    
+    Clear-Line;
+    Write-Host "$('  ' * $Depth)[$operation] " -ForegroundColor Green -NoNewline
+    Write-Host $message -NoNewline
+
+    if (-not $NoNewLine -or $CIBuild) {
+        Write-Host;
+    }
+}
+
+function Undo-Operation($operation, $message, [switch]$NoNewLine, $suppress=$false) {
+    if($suppress) {
+        return;
+    }
+    
+    Clear-Line;
+    Write-Host "$('  ' * $Depth)[$operation] " -ForegroundColor Red -NoNewline
+    Write-Host $message -NoNewline
+
+    if (-not $NoNewLine -or $CIBuild) {
+        Write-Host;
+    }
+}
+
+function Expand-NugetPackages {
+    $foldersToProcess = [System.Collections.Generic.HashSet[string]]@()
+
+    $nupgks = Get-ChildItem *.nupkg
+
+    Push-Depth
+    $nupgks | ForEach-Object {
+        $package = $_.FullName
+        $expand = Join-Path -Path ([System.IO.Path]::GetDirectoryName($package)) -ChildPath ([System.IO.Path]::GetFileNameWithoutExtension($package))
+        ((Test-Path -Path $expand) -and (Remove-Item $expand -Force -Recurse)) | Out-Null
+
+        Update-Operation -operation "extracting" -message "$([System.IO.Path]::GetFileName($package))"
+        [System.IO.Compression.ZipFile]::ExtractToDirectory($package, $expand)
+        Submit-Operation -operation "done" -message "$([System.IO.Path]::GetFileName($package))" -suppress $CIBuild
+
+        $foldersToProcess.Add($expand) | Out-Null
+    }
+    Pop-Depth
+    Write-Host;
+    return $foldersToProcess
+}
+
+function Test-Signatures($rootFolder) {
+    $fail = @{}
+
+    Update-Operation -operation "discovering" -message "Searching binaries..."  -suppress $CIBuild
+    $files = (Get-ChildItem $rootFolder -Recurse -Include "*.dll", "*.exe")
+    Update-Operation -operation "discovered" -message "$($files.Length) binaries discovered"
+
+    $packageId = ([XML](Get-Content $rootFolder\*.nuspec)).package.metadata.id;
+    $unsignedAllowList = @();
+    if ($global:UnsignedAllowList.ContainsKey($packageId)) {
+        $unsignedAllowList = $global:UnsignedAllowList[$packageId];
+    }
+
+    $files | ForEach-Object {
+        $relativePath = Get-RelativePath -path $_.FullName -relativeTo $rootFolder
+        Update-Operation -operation "processing" -message $relativePath -suppress $CIBuild
+
+        $signature = Get-AuthenticodeSignature -FilePath "$($_.FullName)"
+        if ($signature.Status -eq "Valid") {
+            $valid = ($ValidSubjects.Contains($signature.SignerCertificate.Subject) -or $ValidThumbprints.Contains($signature.SignerCertificate.Thumbprint))

```

Comment:
Pass the values by parameters, or use the variable qualification to show that this is coming in externally.